src/core-asyncpipeline/AsyncPipeline/AsyncPipeline.T/ToTask/ToTask.cs

```diff
@@ -0,0 +1,13 @@
+#nullable enable
+
+using System.Threading.Tasks;
+
+namespace System
+{
+    partial struct AsyncPipeline<T>
+    {
+        public Task<T> ToTask()
+            =>
+            isCanceled is false ? valueTask.AsTask() : Task.FromCanceled<T>(CanceledToken());
```

Comment:
I suppose the internal task should not be open to a user*.
It looks like an abstraction leak allows the user to break the pipeline.
I seems the true conversion should be implemented, something like that**:

```cs
isCanceled is false ? InnerToTask(valueTask) : Task.FromCanceled<T>(CanceledToken());

private static async Task<T> InnerToTask(ValueTask<T> valueTask)
    =>
    await valueTask.ConfigureAwait(false);
```

*Even being wrapped to Task. Also, see [the sources](https://github.com/dotnet/corert/blob/master/src/System.Private.CoreLib/shared/System/Threading/Tasks/ValueTask.cs) ValueTask.ToTask() may not only wrap ValueTask to Task but also may extract the internal Task object if present - so, an abstraction leak can be made for the pipeline.

**Also it can be implemented by using the `InnerToValueTask` (see the comment for the `ToValueTask()` issue).
(But I think the first solution with the own `InnerToTask` looks better.)
```cs
isCanceled is false ? InnerToValueTask(valueTask).AsTask() : Task.FromCanceled<T>(CanceledToken());
```